### DISCUSSION TITLE
Temporary Admin User - Create Bootstrap User Even With Permanent Admins in Database

### DISCUSSION THREAD

**Discussion by ncphins** [keycloak/keycloak#35112] [Archive]
Maybe this is intentional, but I wanted to ask. I am migrating our installs and install guides to 26.x.x. I noticed that it was pretty easy to create an admin user using the bootstrap-admin command, even on an install that has permanent admins setup.

1. I started a fresh database.
2. I ran the bootstrap-admin command to bootstrap my admin user for the setup. 
3. I logged in, setup the server, and created a new permanent admin.
4. I deleted the bootstrapped admin user. 
5. All was fine. But then, I accidentally ran the bootstrap-admin command again, and the server was more than happy to create the bootstrapped admin for me, allowing me to login as the admin I created.

I realize it requires access to the machine, but it seems like an easy way to create an admin on the server. Is this intentional behavior? I assumed it was but it seemed like it at least warranted a question. Maybe I am simply not understanding the purpose of the bootstrapped admins.

**Comment by ncphins** [comment]
This is starting to feel like a bug to me. The behavior is different if I create the temporary admin user with the ENV variable  KC_BOOTSTRAP_ADMIN_USERNAME. In that case, it does not recreate the admin user. In fact, that appears to only work on a fresh database on the first startup. 

I believe I will transfer this discussion to an issue.

**Reply by shawkins** [reply]
> Is this intentional behavior?

Yes this is intentional. 

> In fact, that appears to only work on a fresh database on the first startup.

That is also the desired behavior.

**Comment by ncphins** [comment]
I reopened this to provide the information Steven provided. This has been verified as the expected behavior. Now, as I read the doc https://www.keycloak.org/server/bootstrap-admin-recovery it is clearer how this is supposed to work. I appreciate the help.
